cloud.foundry.cli/src/main/java/cloud/foundry/cli/logic/ApplyLogic.java

```diff
@@ -51,31 +58,68 @@ public void applyApplications(@Nonnull Map<String, ApplicationBean> desiredAppli
 
         ApplicationsOperations applicationsOperations = new ApplicationsOperations(cfOperations);
         log.info("Fetching information about applications...");
-        Map<String, ApplicationBean> liveApplications = applicationsOperations.getAll().block();
+        Map<String, ApplicationBean> liveApplications = this.getLogic.getApplications(applicationsOperations);
         log.info("Information fetched.");
 
         // that way only the applications of the live system are compared in the diff
         ConfigBean desiredApplicationsConfig = createConfigFromApplications(desiredApplications);
         ConfigBean liveApplicationsConfig = createConfigFromApplications(liveApplications);
 
         // compare entire configs as the diff wrapper is only suited for diff trees of these
-        DiffLogic diffLogic = new DiffLogic();
         log.info("Comparing the applications...");
-        DiffResult wrappedDiff = diffLogic.createDiffResult(liveApplicationsConfig, desiredApplicationsConfig);
+        DiffResult diffResult = this.diffLogic.createDiffResult(liveApplicationsConfig, desiredApplicationsConfig);
         log.info("Applications compared.");
 
-        Map<String, List<CfChange>> allApplicationChanges = wrappedDiff.getApplicationChanges();
-        ApplicationsOperations appOperations = new ApplicationsOperations(cfOperations);
+        Map<String, List<CfChange>> allApplicationChanges = diffResult.getApplicationChanges();
 
         Flux<Void> applicationRequests = Flux.fromIterable(allApplicationChanges.entrySet())
-                .flatMap( appChangeEntry -> ApplicationRequestsPlaner.create(appOperations, appChangeEntry.getKey(),
+                .flatMap( appChangeEntry -> ApplicationRequestsPlaner.create(applicationsOperations,
```

Comment:
I would rename the method 
"create". Because create sounds a bit to generic. 

Maybe createServiceRequest or something like that.

Reply:
Changed by chris.